### PR TITLE
Remove unused vars and improve code readability

### PR DESCRIPTION
**Pull request by arun1587**
hey, was able to make use of linters to fix some code there by improving the readability a bit. 
imo, these changes shouldn't have functional or performance impact.

### CODE REVIEW

File: message/option.go [hunk 1]
```diff
@@ -243,13 +243,13 @@ func extendOpt(opt int) (int, int) {
 func marshalOptionHeaderExt(buf []byte, opt, ext int) (int, error) {
 	switch opt {
 	case ExtendOptionByteCode:
-		if buf != nil && len(buf) > 0 {
```
Comment (arun1587) [inline message/option.go:246]: length of a nil buffer would be `0` so `buf !=nil` check is redundant.

File: message/options.go [hunk 2]
```diff
@@ -558,7 +558,7 @@ func (options Options) Clone() (Options, error) {
 	opts, used, err := opts.ResetOptionsTo(buf, options)
 	if err == ErrTooSmall {
 		buf = append(buf, make([]byte, used-len(buf))...)
-		opts, used, err = opts.ResetOptionsTo(buf, options)
```
Comment (arun1587) [inline message/options.go:561]: `used` var isn't really used anywhere beyond this line. hence it can be omitted.

File: net/dtlslistener.go [hunk 3]
```diff
@@ -162,13 +162,11 @@ func (l *DTLSListener) Accept() (net.Conn, error) {
 	}
 
 	if deadline.IsZero() {
-		select {
```
Comment (arun1587) [inline net/dtlslistener.go:165]: use of select for a single case is unnecessary.

File: mux/middleware.go [hunk 4]
```diff
@@ -5,19 +5,12 @@ package mux
...
-}
-
 // Middleware allows MiddlewareFunc to implement the middleware interface.
 func (mw MiddlewareFunc) Middleware(handler Handler) Handler {
 	return mw(handler)
 }
 
 // Use appends a MiddlewareFunc to the chain. Middleware can be used to intercept or otherwise modify requests and/or responses, and are executed in the order that they are applied to the Router.
 func (r *Router) Use(mwf ...MiddlewareFunc) {
-	for _, fn := range mwf {
```
Comment (arun1587) [inline mux/middleware.go:20]: remove unnecessary for loop

**Review by jkralik — APPROVED** [review]
Great and thanks for contributing.

**Pull request merged**